lib/zinnia.js

```diff
@@ -220,91 +312,33 @@ export async function run ({
     })
   }
 
-  let shouldRestart
+  const shouldRestart = {
+    value: false,
+    set (value) {
+      this.value = value
+    },
+    get () {
+      return this.value
+    }
+  }
```

Comment:
@bajtos this is new: I turned `shouldRestart` into an object, so that functions that get a reference to it can still change its value.

Reply:
Yeah, I noticed. I have mixed feelings about this structure, but I also cannot come up with an option that would be clearly better. Let's keep your version for now. We can relatively easily refactor this place if we discover a better approach.

Reply:
I share your feelings. Sounds good!